pravega/zookeeper-operator
# Issue 63: Added support for ephemeral storage

Signed-off-by: prabhaker24 <prabhaker.saxena@dell.com>

### Change log description
This pr is to enable zookeeper cluster to be able to use ephemeral storage, At any point, only one storage can be there either ephemeral or persistent and if none is specified zookeeper cluster will use persistent storage.

### Purpose of the change
fixes #63 

### What the code does
This PR allows users to create zookeepers with ephemeral storage instead of PVs.

To enable ephemeral storage add this to the manifest,

```
spec:
    storageType: ephemeral   
```

### How to verify it
1) Create a ZK cluster with storageType as Ephemeral, in this case, Ephemeral storage should get created
2) Create a ZK cluster with storageType as persistence, in this case, persistence storage should get created
3) Create a ZK cluster with no storageType, in this case, Persistence storage should get created.
4) Try upgrading the operator keeping the ZK cluster same operator should be able to reflect the correct persistence configuration post-upgrade.
Note:- In case of ephemeral storage, the cluster may not be able to come back up if more than quorum number of nodes are restarted simultaneously.
I have verified all the above 4 scenarios
